typeobjectpattern/README.md

```diff
@@ -13,8 +13,72 @@ As explained in the book Game Programming Patterns by Robert Nystrom, type objec
 > Allowing flexible creation of new “classes” by creating a single class, each instance of which represents a different type of object
 
 ## Explanation
-Say, we are working on a game which has a hero and many monsters which are going to attack the hero. These monsters have certain attributes like attack, points etc. and come in different 'breeds' like zombie or ogres. The obvious answer is to have a base Monster class which has some fields and methods, which may be overriden by subclasses like the Zombie or Ogre class. But as we continue to build the game, there may be more and more breeds of monsters added and certain attributes may need to be changed in the existing monsters too. The OOP solution of inheriting from the base class would not be an efficient method in this case.
-Using the type-object pattern, instead of creating many classes inheriting from a base class, we have 1 class with a field which represents the 'type' of object. This makes the code cleaner and object instantiation also becomes as easy as parsing a json file with the object properties.
+
+Real-world example
+
+
+>Say, we are working on a game which has a hero and many monsters which are going to attack the hero. These monsters have certain attributes like attack, points etc. and come in different 'breeds' like zombie or ogres. The obvious answer is to have a base Monster class which has some fields and methods, which may be overriden by subclasses like the Zombie or Ogre class. But as we continue to build the game, there may be more and more breeds of monsters added and certain attributes may need to be changed in the existing monsters too. The OOP solution of inheriting from the base class would not be an efficient method in this case.
+>Using the type-object pattern, instead of creating many classes inheriting from a base class, we have 1 class with a field which represents the 'type' of object. This makes the code cleaner and object instantiation also becomes as easy as parsing a json file with the object properties.
+
+
+In plain words
+
+>It's natural to identify the type of an object with its class, but this can lead to unwieldy inheritance hierarchies or inheritance hierarchies with multiple representations of the same object. This, in turn, can lead to synchronization problems.
+
+
+**Programmatic Example**
+
+With that game design in mind, we fire up our text editor and start coding. According to the design, a dragon is a kind of `monster`, a troll is another kind, and so on with the other breeds. Thinking object-oriented, that leads us to a `Monster` base class:
+
+```java
+class Monster
+{
+public:
+  virtual ~Monster() {}
+  virtual const char* getAttack() = 0;
+
+protected:
+  Monster(int startingHealth)
+  : health_(startingHealth)
+  {}
+
+private:
+  int health_; // Current health.
+};
```

Comment:
I'm sorry, but this doesn't look like Java. Please reference the code in this repository.